libs/gpg45-evaluator/src/main/java/uk/gov/di/ipv/core/library/gpg45/Gpg45ProfileEvaluator.java

```diff
@@ -200,20 +197,34 @@ && doesEvidenceContainEvidenceType(evidenceItem, evidenceType)) {
         return Optional.empty();
     }
 
-    public int calculateF2FRequiredStrengthScore(List<RequiredGpg45ScoresDto> requiredGpg45Scores) {
-        int strengthScore = 0;
-        for (RequiredGpg45ScoresDto gpg45Score : requiredGpg45Scores) {
-            Gpg45ScoresDto requiredScores = gpg45Score.getRequiredScores();
-            if (requiredScores.getFraud() > 0 || requiredScores.getActivity() > 0) {
-                continue;
-            }
-            List<EvidenceDto> evidences = requiredScores.getEvidences();
-            if (!evidences.isEmpty()
-                    && (strengthScore == 0 || evidences.get(0).getStrength() < strengthScore)) {
-                strengthScore = evidences.get(0).getStrength();
-            }
-        }
-        return strengthScore;
+    public List<List<Gpg45Scores.Evidence>> calculateEvidencesRequiredToMeetAProfile(
```

Comment:
I think it might worth extending this for completeness, since it shouldn't be too hard to do and makes it a lot more reusable - at the moment the signature of this method is a bit weird.

I also wonder if this would be more convenient as an instance method on `GPG45Scores`? i.e. so that it becomes:

```
public List<GPG45Scores> calculateRequiredScoresToMeetAProfile(List<GPG45Profile> profiles) { .. }

// In F2F call
var requiredScores = gpg45Scores.calculateRequiredScoresToMeetAProfile(ACCEPTED_PROFILES);
```

I don't think the implementation would need to change much - we'd just need to add the activity/fraud/verification scores to this block?

Reply:
In any case I think you have to check activity/fraud/verification scores _somewhere_ - at the moment this isn't actually going to work.

In particular, the existing behaviour is deciding whether you need:
- strength 4 (to match M1A with a fraud score of 1)
- strength 3 (to match M1B with a fraud score of 2)

Since you're ignoring the fraud score, we'll always say that strength 3 is good enough - but it isn't if you only have a fraud score of 1!